### PR TITLE
Add command line tests

### PR DESCRIPTION
**Pull request by Julow**
So we can avoid regressions.

Also adds a few tests for invalid combinations of options.
It's not complete, there is a lot more to test.

### CODE REVIEW

File: test/cli/gen/gen.ml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      update_or_add test_name ~f:(fun e -> { e with should_fail = true })
+  | _ ->
+      entries
+
+let check_test test_name entry ok =
+  let e ok b =
+    if b then Format.kfprintf (fun _ -> false) Format.err_formatter
+    else Format.ikfprintf (fun _ -> ok) Format.err_formatter
+  in
+  let ok = e ok (not entry.has_ref) "@{<error>Error@}: Missing file %s.ref\n" test_name in
```
Comment (emillon) [inline test/cli/gen/gen.ml]: I think you can simplify this part by using `invalid_arg` as it's only used to exit with an error message

File: test/cli/dune [hunk 2]
```diff
@@ -0,0 +1,10 @@
+(include dune.inc)
```
Comment (emillon) [inline test/cli/dune]: dune file generation is a bit heavy for these kind of tests, I'll try to refactor this in a future PR.

File: test/cli/gen/gen.ml (hunk 1 above)
Comment (Julow) [reply test/cli/gen/gen.ml]: I did that way to print all the warnings before exiting. Do you think I should remove that feature ?

File: test/cli/dune (hunk 2 above)
Comment (Julow) [reply test/cli/dune]: You mean it's slowing dune ? `dune runtest` already take a few seconds.

Comment (emillon) [reply test/cli/dune]: I mean that it's a bit heavy to have arguments in a file, output in another file, etc. I'll try to see if that's better with a different kind of test, but this works in the meantime.

Comment (Julow) [reply test/cli/dune]: I see. `gen.ml` could read a sexp file, what do you think ?

Comment (emillon) [reply test/cli/dune]: let me work something out in another PR, the most important part is that we now have tests!

**Comment by emillon** [comment]
Thanks!

**Pull request merged**